flocker/acceptance/endtoend/test_dockerplugin.py

```diff
@@ -48,6 +50,25 @@ def get_path(name):
         return Client(base_url="https://{}:{}".format(address, DOCKER_PORT),
                       tls=tls, timeout=100)
 
+    def restart_docker(self, address):
+        """
+        Restart the Docker daemon on the specified address.
+
+        :param bytes address: The public IP of the node on which Docker will
+            be restarted.
+        """
+        try:
+            distro = run_ssh_command(
+                b"root", address, ["python", "-m", "platform"])
+            distro = distro.output.lower()
+            if 'ubuntu' in distro:
+                command = ["service", "docker", "restart"]
+            else:
+                command = ["systemctl", "restart", "docker"]
+            run_ssh_command(b"root", address, command)
```

Comment:
I don't have any good ideas here. I would have thought that however we connect to the machines, we would be able to use `sudo`.